docs/design-proposals/tidb-stable-scheduling.md

```diff
@@ -0,0 +1,167 @@
+    + [Workarounds](#workarounds)
+  * [tidb-cluster cannot schedule new pod of TiDB member back to its node if it does not meet new requirements](#tidb-cluster-cannot-schedule-new-pod-of-tidb-member-back-to-its-node-if-it-does-not-meet-new-requirements)
+    + [Workarounds](#workarounds-1)
+
+## Glossy
+
+- Pod: A deployable object in Kubernetes.
+- Node: A node is worker machine to run pods in Kubernetes.
+- TiDB cluster: TiDB cluster is database cluster which is constructed by TiDB
+  server, PD server and TiKV server. Each TiDB/PD/TiKV server is a cluster too
+  which consist of one or more members.
+- TiDB server: One of key components of TiDB cluster. It's access point of TiDB
+  cluster.
+- TiDB member: member of TiDB server which holds a unique network identifier.
+- TiDB pod: for each TiDB member, there is at most one running pod. It may
+  crash or be replaced by a new pod.
+- Load balancer: A component which proxies traffic for applications, e.g. LVS,
+  HAProxy, F5, etc.
+
+## Motivation
+
+There are use cases that users need to access TiDB server from outside the
+Kubernetes cluster. But in some environments (e.g. on bare-metal machines), we
+may lack load balancer solution or need to configure IP addresses of TiDB
+services in existing load balancer.
+
+In this scenario, we need to use `NodePort` service. By default, NodePort
+services are available cluster-wide on all nodes but cannot propagate client's
+IP address to the end pods and may cause a second hop. 
+
+For propagating client's IP address and better performance, we can
+set `externalTrafficPolicy` of service to `Local`. A side-effect is the service of
+TiDB will be accessible only on the nodes which a running TiDB pod. To avoid
+manual intervention to update IP addresses in load balancer when performing
+a rolling update, we prefer to schedule new pod of TiDB member to its previous node.
+
+## Goals
+
+- Able to schedule new pod of TiDB member to its previous node
+
+### Non-Goals
+
+- Stable scheduling for TiKV/PD pods
+- Guarantee that new pod of TiDB member will be scheduled to its node if
+  there is another scheduler in cluster which may schedule pods to its node
+- Guarantee that new pod of TiDB member will be scheduler to its node if some
+  scheduling requirements changed
+
+## Use Cases
+
+### No need to update IP addresses of TiDB in load balancer outside of the Kubernetes cluster
+
+When a TiDB cluster is running in a dedicated Kubernetes cluster or nodes of
+TiDB cluster are reserved for it. After a rolling update of TiDB cluster is
+done, new pods of TiDB member will be scheduled to theirs previous nodes. User
+does not need to update IP addresses in load balancer.
+
+## Proposal
+
+Currently, we have tidb-scheduler to schedule all pods of TiDB cluster. We can
+write a new predicate function for pods of TiDB server. In this new predicate
+function we can choose the its previous node of TiDB member it it exists in candidate
```

Comment:
In this new predicate function we can choose the previous node of TiDB member if it exists in candidate nodes.